Classes/ViewHelpers/FormViewHelper.php

```diff
@@ -58,10 +67,15 @@ protected function getFormObjectName()
     protected function getFormActionUri()
     {
         /** @var ActionRequest $actionRequest */
-        $actionRequest = $this->controllerContext->getRequest();
-        $uri = $actionRequest->getHttpRequest()->getUri();
-        if ($this->hasArgument('section') && $this->arguments['section'] !== '') {
-            $uri = $uri->withFragment($this->arguments['section']);
+        $actionRequest = clone $this->controllerContext->getRequest();
+        $requestUri = $actionRequest->getHttpRequest()->getUri();
+        $uri = $this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest()
```

Comment:
The `$fallbackRequest` should be specified for good measure:
```suggestion
        $uri = $this->baseUriProvider->getConfiguredBaseUriOrFallbackToCurrentRequest($httpRequest)
```